manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/UsingTest.scala

```diff
@@ -19,61 +19,140 @@
+      section("Query using an index hint") {
+        p("""The query above will not naturally pick an index to solve the plan.
+            |This is because the graph is very small, and label scans are faster for small databases.
+            |In general, however, query performance is ranked by the dbhit metric, and we'll see that using an index is
+            |slightly better for this query.""")
+        query(s"$matchString USING INDEX liskov:Scientist(name) RETURN liskov.born AS $columnName",
+              assertIntegersReturned(1939)) {
+          p("Returns the year Barbara Liskov was born.")
+          profileExecutionPlan()
+        }
+      }
+      section("Query using multiple index hints") {
+        p("""Supplying one index hint changed the starting point of the query, but the plan is still 'straight', i.e. it has
+            |only one starting point. If we give the planner yet another index hint, we force it to use two starting points,
+            |one at each end of the match. It will then join these two branches using a join operator. """)
+        query(s"$matchString USING INDEX liskov:Scientist(name) USING INDEX conway:Scientist(name) RETURN liskov.born AS $columnName",
+              assertIntegersReturned(1939)) {
+          p("Returns the year Barbara Liskov was born, using a slightly better plan.")
+          profileExecutionPlan()
+        }
+      }
+    }
+    section("Scan hints") {
+      p("""If your query matches large parts of an index, it might be faster to scan the label and filter out nodes that do not match.
+          |To do this, you can use `USING SCAN variable:Label` after the applicable `MATCH` clause.
+          |This will force Cypher to not use an index that could have been used, and instead do a label scan.""")
+      section("Hinting a label scan") {
+        p("""If the best performance is to be had by scanning all nodes in a label and then filtering on that set, use `USING SCAN`.""")
+        query(s"""MATCH (s:Scientist)
+                 |USING SCAN s:Scientist
+                 |WHERE s.born < 1939
+                 |RETURN s.born AS $columnName""",
+              assertIntegersReturned(1938, 1867)) {
+          p("Returns all scientists born before 1939.")
+          profileExecutionPlan()
+        }
+      }
+    }
+    section("Join hints") {
+      p("""Join hints are the most advanced type of hints, and are not used to find starting points for the
+          |query execution plan, but to enforce that joins are made at specified points. This implies that there
+          |has to be more than one starting point (leaf), in order for the query to be able to join the two branches ascending
+          |from these leaves. Due to this nature, joins, and subsequently join hints, do have the potential to force
+          |the planner to pick a very bad starting point, which will negatively affect query performance. In other cases,
```

Comment:
I rephrased to `This implies that there has to be more than one starting point (leaf) in the plan, in order for the query to be able to join the two branches ascending from these leaves. Due to this nature, joins, and subsequently join hints, will force the planner to look for additional starting points, and in the case where there are no more good ones, potentially pick a very bad starting point. This will negatively affect query performance.`